### ISSUE TITLE
No such file or directory 7z on MacOS

### ISSUE THREAD

**Issue by drewtua** [ciromattia/kcc#730]
Hello guys,

Can anybody help me?
Every time i try to convert a cbr to mobi I get "No such file or directory 7z"

<img width="513" alt="Captura de Pantalla 2024-08-09 a las 22 46 41" src="https://github.com/user-attachments/assets/06eed194-557f-47ce-b4af-8234d6ba5ec0">

Thanks

**Comment by axu2** [comment]
The error message is misleading, the error message will be clearer in the future.

It probably means that the extraction failed for some reason, probably because the file is bad somehow.

Can you open a terminal and type:

`tar -xf '/Users/andreagarciatua/Desktop/Nana - Tomo 03 (#005-008).cbr'`

and tell me what it says?

**Comment by drewtua** [comment]
> tar -xf '/Users/andreagarciatua/Desktop/Nana - Tomo 03 (#005-008).cbr'

Thanks!
I got this

<img width="561" alt="Captura de Pantalla 2024-08-09 a las 23 03 08" src="https://github.com/user-attachments/assets/1fad2b52-29b5-48f8-8a2a-4303a2e8cc2b">

**Comment by axu2** [comment]
wow, so `tar` failed.

Can you try using 6.0.0 instead, and following the instructions to install 7z and unar? @drewtua

**Comment by drewtua** [comment]
> wow, so `tar` failed.
> 
> Can you try using 6.0.0 instead, and following the instructions to install 7z and unar? @drewtua

I already tried it with other versions and got the same message

**Comment by axu2** [comment]
Can you please provide a screenshot of the whole UI of 6.0.0 with the error message that looks like this?

![image](https://github.com/user-attachments/assets/4a19c294-07fc-463d-b7fd-9b94eb9a02da)

**Comment by drewtua** [comment]
> Can you please provide a screenshot of the whole UI of **kcc 6.0.0** with the error message that looks like this? @drewtua
> 
> ![image](https://private-user-images.githubusercontent.com/20757319/356755924-4a19c294-07fc-463d-b7fd-9b94eb9a02da.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMyMzg3MjYsIm5iZiI6MTcyMzIzODQyNiwicGF0aCI6Ii8yMDc1NzMxOS8zNTY3NTU5MjQtNGExOWMyOTQtMDdmYy00NjNkLWI3ZmQtOWI5NGViOWEwMmRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA5VDIxMjAyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4MmRiMzdlM2Y1NzkyOGE3NGY2NTk0OGY2OWE1MWJjODRkMzAxZDQ0MDkzNDEzYjMzZGQ3YTAwNThlMWU1NmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4wjgmfW-Bum8QcANc2skVoCcWpFSV-FX0MGqEKBXWhQ)

With 6.0 I get this error

<img width="512" alt="Captura de Pantalla 2024-08-09 a las 23 20 16" src="https://github.com/user-attachments/assets/0ab12ebe-f49a-4278-b91a-83099e76fe88">

**Comment by axu2** [comment]
Did KCC not ask you to install 7z and unar when you opened it? try again after following these instructions. https://github.com/ciromattia/kcc/wiki/Installation#7-zip

**Comment by drewtua** [comment]
> Did KCC not ask you to install 7z and unar when you opened it? try again after following these instructions in 6.0. https://github.com/ciromattia/kcc/wiki/Installation#7-zip

I tried to install it, but idk how in macOS

**Comment by axu2** [comment]
```zsh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
brew install p7zip
brew install unar
```

**Comment by drewtua** [comment]
I swear my computer hates me

<img width="634" alt="Captura de Pantalla 2024-08-09 a las 23 40 40" src="https://github.com/user-attachments/assets/91cbd461-3924-4a0a-a2e1-63a038f30213">

**Comment by axu2** [comment]
I'd try again, maybe copy the command directly from the website. https://brew.sh/

I can open the first errored url just fine https://formulae.brew.sh/api/formula.jws.json

Can you open that link? What macOS version are you on? Is any software that might block certain website running on it?

**Comment by drewtua** [comment]
> I'd try again, maybe copy the command directly from the website. https://brew.sh/


I get the same error

**Comment by drewtua** [comment]
> Can you open that link? What macOS version are you on? Is any software that might block certain website running on it?

Yeah, I can open it!
Monterrey 12.7.6

**Comment by axu2** [comment]
OK, let's hop on discord then. https://discord.com/invite/qj7wpnUHav

**Comment by drewtua** [comment]
I finally could install Homebrew and downloaded 6.0 of KCC and it's working!!!

Thank you so much for all your help, axu2!!

**Comment by axu2** [comment]
OK, I'll fix it again in 6.2

**Comment by axu2** [comment]
Went ahead and build v6.1.1 https://github.com/ciromattia/kcc/releases/tag/v6.1.1

Please confirm it fixes the issue.
